pkg/logs/logs.go

```diff
@@ -96,10 +83,15 @@ func Start() error {
 	atomic.StoreInt32(&isRunning, 1)
 	log.Info("logs-agent started")
 
-	// add the default sources
-	for _, source := range config.DefaultSources() {
-		sources.AddSource(source)
-	}
+	// add the default sources after the AutoConfig has been runned once because
+	// we don't want the container_collect_all to reap all containers ownership
+	go func() {
+		common.BlockUntilAutoConfigRunnedOnce(time.Second * 30)
```

Comment:
I'm afraid 30 seconds could be a lot, we should think about short lived containers that we could miss in 30 seconds. Could we make t configurable at least?

Reply:
Interesting.
This is a timeout because I don't want to have a `for {}` running forever in the case of an issue (bug) in the autoconfig.
In the normal behavior of an Agent startup, this timeout won't occur and this will probably be blocking only for hundreds of milliseconds (the time needed for the autoconfig to run). With this in mind, do you think this timeout should still be configurable? I've nothing against a new configuration field (except from the fact that the Agent has already a lot ;D), I just want to be sure that these 30s are properly understood.

Reply:
yes we're on the same page, of course we should keep the timeout as protection. I'm just saying 30s as default timeout could be a lot in case of an issue, if we know that the nominal case should only takes < 1 second I guess we can set a 5s timeout no?
Taking into consideration this is only executed at the agent start/restart, it's fait to say it's not very likely to have the timeout reached + a short lived container existed at the same time but theoretically it could happen. Having a shorter timeout would reduce this risk considerably.

Reply:
I’m also wondering under which conditions the timeout will be hit.
Right now, if the AutoConfig hasn’t run once before the timeout, we continue as if it successfully ran (And we log a warning)
I’m wondering what’s best between continuing in best effort mode (ignoring the fact the AutoConfig didn’t ran) and making the issue “more visible”.

The reason why I’m not a big fan of too aggressive timeouts is because in cases where the agent run slower than usual (because it is running on a heavily loaded host or because it is running with instrumentation features that are slowing it down (like valgrind)), the agent will not run slower, it will also behave differently.

Reply:
First, @L3n41c you made me realize that throwing an error is better than throwing a warning here, I'll change this, it'll be a bit more visible. Maybe we should (in another PR) add the autoconfig in the health mechanism of the Agent, WDYT?

As mentioned by @L3n41c this lenient timeout is to give some space to the AutoConfig to run on loaded hosts. I guess using a configurable field won't hurt anyways, let me add that.

Reply:
Also I should mention: even if the AutoConfig takes longer, it won't stop it for loading, it'll just start doing this (loading some default sources in the logs agent) before the AutoConfig has finished to prepare.